.internal-ci/charts/reserve-auditor/templates/ingress.yaml

```diff
@@ -1,61 +1,31 @@
-{{- if .Values.ingress.enabled -}}
-{{- $fullName := include "reserve-auditor.fullname" . -}}
-{{- $svcPort := .Values.service.port -}}
-{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
-  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
-  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
-  {{- end }}
-{{- end }}
-{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1
-{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
-apiVersion: extensions/v1beta1
-{{- end }}
 kind: Ingress
 metadata:
-  name: {{ $fullName }}
+  name: {{ include "reserve-auditor.fullname" . }}
   labels:
     {{- include "reserve-auditor.labels" . | nindent 4 }}
-  {{- with .Values.ingress.annotations }}
   annotations:
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+    {{- toYaml .Values.ingress.annotations | nindent 4 }}
```

Comment:
I see what @jgreat was probably suggesting you do here.  But if you don't have ANY annotations, then this YAML bloc will fail.  I'm not suggesting that you change it back to what it was, just pointing out what the `{{- with .Values.ingress.annotations }}` makes sense in this case.

Reply:
Also, @jgreat and I disagree on the usage of the `{{- .with }}` pattern anyway.  You don't need to be caught in the middle of it ;)

Reply:
I had the same potential concern I mentioned to Ricky in-passing the other day. Personally don't mind using the `with` for that use-case (or imo even just to  change context when there are a bunch of nested values).

I also think if we want to establish changes from the default template we should create our own starter chart to pass to `helm create --starter ..`. Without that, I question whether manually making these changes on every chart is worthwhile.

Reply:
The direct "toYaml" won't make it fail, as long as there is at least `annotations: {}` in the `values.yaml`. In our internal charts where will always be some kind of value defined.

The `with` pattern just adds unnecessary noise when there's only one value referenced and all that bracket noise adds confusion to casual chart contributors. 

My opinion is, where the consumers of a chart are only internal, we should have very direct, almost static manifests.  We want to empower the development teams to contribute and minimizing the tempting/logic/abstraction is a way we can reduce the barrier to entry.